docs/build/build-options.md

```diff
@@ -421,6 +422,55 @@ to use either of those (`ENABLE_STDCXX_SYNC` excludes `ENABLE_MONOTONIC_CLOCK`).
 When ON, this option enables the standard C++ `thread` and `chrono` libraries 
 (available since C++11) to be used by SRT instead of the `pthreads` libraries.
 
+#### ENABLE_PKTINFO
+**`--enable-pktinfo`** (default: OFF)
+
+This allows the use of `IP_PKTINFO` control message which can be used to
+extract the true target IP address from the incoming UDP packets to a listener
+bound to a so-called "any" address (`0.0.0.0` for IPv4, AKA `INADDR_ANY`, or
+`::` for IPv6, AKA `in6addr_any`, or simply when the application clears the
+`sockaddr*` structure and only sets the port number), and then set this address
+as source IP address when sending packets to the peer. This solves the problem,
+when the routing rules in the agent's host make that sending a packet to a
+given IP address would use a different source IP address than the one that was
+set as a target IP address in the UDP packet from the peer. The peer will
+reject such a packet as a suspected MITM attempt, and this will lead to a
+connection failure.
+
+This problem has been observed in the case, when the agent's host has at least
+two IP addresses that share the same broadcast prefix and it is being contacted
+by the peer using other than the first of these addresses. For example:
+
+The host has set the following local IP addresses:
+
+* 192.168.10.5 - routing to 192.168.10.1
+* 10.10.5.10 - routing to 10.10.5.1
+* 10.10.5.20 - routing to 10.10.5.1
+
+In all of them the netmask is 255.255.255.0, which means that the second
+and third IP address share the same broadcast prefix, which is 10.10.5.0.
+The problematic case is when an agent running on this host will be contacted
+by a peer using the address 10.10.5.20.
+
+In such a case the source address set in the UDP packet sent back to the peer
+will always be the first of these addresses (10.10.5.10), which will be
+different than the one the peer used to contact (10.10.5.20). The peer will
+then rejeect such a packet by having the source address different than the one
+it tries to contact.
```

Comment:
```suggestion
In such a case the source address set in the UDP packet being sent
will always be the first of these addresses (10.10.5.10), which will be
different from the one from which the packet is actually being sent (10.10.5.20). The peer will
then reject such a packet because its source address is different.
```

Reply:
, which will be different from the one TO which the (incoming) packet is actually being sent.

To depict what the problem really is, that's more-less:

[caller]  UDP packet,  SRC:caller, DEST:10.10.5.20  -> send to  [listener]

[listener]  UDP packet, SRC:10.10.5.10, DEST:caller -> [caller]

[caller] Weird, I was sending to 10.10.5.20, not to 10.10.5.10, I don't know this address! Reject!

The fix is undertaken on the [listener] and relies on reading the value from the DEST address (10.10.5.20) and then when sending the packet back to the [caller], this very address is set forcefully as SRC.

Reply:
```suggestion
In such a case the source address set in the UDP packet being sent will 
always be the first of these addresses (10.10.5.10), which will be different 
from the one TO which the (incoming) packet is actually being sent (10.10.5.20). 
The peer will then reject such a packet because its source address is different.
This interaction is shown in the following example:

    [caller] UDP packet, SRC:caller, DEST:10.10.5.20 -> send to [listener]
    [listener] UDP packet, SRC:10.10.5.10, DEST:caller -> [caller]
    [caller] I was sending to 10.10.5.20, not to 10.10.5.10, I don't know this address! Reject!

Note that the fix is undertaken on [listener] and relies on reading the value from the DEST address (10.10.5.20). When sending the packet back to the [caller], this address is set forcefully as SRC.
```